### PR TITLE
Testing/error handling

### PR DESCRIPTION
**Pull request by KatherineBlaine**
### Files Changed:

- App.js
- NotFound.js
- error-handling-spec-cy.js

### What Has Changed: 

- Added return home button on not found page
-  Added a link in the NotFound component and passed a method to NotFound to reset the error prop when the user returns home so the page will re render!
- Stubbed GET, POST, and PUT requests with error status codes
- Tested that the not found page rendered properly and the user was able to return home when seeing this page

### Notes: (includes questions, ideas for the future, anything you want to call out)

Don't think theres anything else to add! That should be all for testing!!

### CODE REVIEW

File: client/cypress/e2e/error-handling-spec.cy.js [hunk 1]
```diff
@@ -1,5 +1,97 @@
...
+
+describe('400 POST error handling', () => {
+    beforeEach(() => {
+      cy.intercept('GET', 'http://localhost:3001/api/v1/entries', {
+        statusCode: 200,
+        fixture: 'entries.json'
+      })
+      cy.intercept('POST', 'http://localhost:3001/api/v1/entries', {
+        statusCode: 400,
+    })
```
Comment (jfogiato) [inline client/cypress/e2e/error-handling-spec.cy.js:37]: Awesome!! Thank you for being so thorough with this testing KB!

File: client/cypress/e2e/error-handling-spec.cy.js [hunk 2]
```diff
@@ -1,5 +1,97 @@
...
+
+  it('Should display a home button that can be clicked to move back to the home page', () => {
+    cy.get('.uni-btn').should('be.visible').contains('HOME').click()
+    cy.url().should('eq', 'http://localhost:3000/')
+    cy.get('.home').should('be.visible')
+  })
+})
+
+
+
```
Comment (espressoGoddess) [inline client/cypress/e2e/error-handling-spec.cy.js:97]: So thorough! great work

File: client/src/Components/App/App.js [hunk 3]
```diff
@@ -21,6 +21,10 @@ const App = () => {
   const [filteredActivities, setFilteredActivities] = useState([]);
   const [error, setError] = useState('');
 
+  const resetError = () => {
+    setError('')
```
Comment (espressoGoddess) [inline client/src/Components/App/App.js:25]: good call on resetting the error

File: client/src/Components/NotFound/NotFound.js [hunk 4]
```diff
@@ -19,5 +21,6 @@ const NotFound = ({err}) => {
 export default NotFound;
 
 NotFound.propTypes = {
-  err: PropTypes.object
+  err: PropTypes.object, 
+  resetError: PropTypes.func
```
Comment (espressoGoddess) [inline client/src/Components/NotFound/NotFound.js:25]: heck yea, thanks for adding the proptypes

**Review by espressoGoddess — APPROVED** [review]

File: client/cypress/e2e/error-handling-spec.cy.js [hunk 5]
```diff
@@ -1,5 +1,97 @@
...
+    cy.get('.notfound-header').should('be.visible').contains('Oh no! This page either moved or doesn\'t exist. Try something else.')
+  })
+
+  it('Should display the not found blurb messages', () => {
+    cy.get('.notfound-blurb').should('be.visible')
+    cy.get('.notfound-blurb > :nth-child(1)').should('be.visible').contains('P.S. - Hope you\'re doing well. 🤍')
+    cy.get('.notfound-blurb > :nth-child(2)').should('be.visible').contains('P.P.S. - Proud of you.')
+  })
+
+  it('Should display a home button that can be clicked to move back to the home page', () => {
```
Comment (chrissycooper) [inline client/cypress/e2e/error-handling-spec.cy.js:89]: love it, really great for UX. I could see it being very frustrating to be stuck there, thanks for adding the button and immediately adding a text for it

File: client/src/Components/App/App.js [hunk 6]
```diff
@@ -66,7 +70,7 @@ const App = () => {
   return (
     <main>
       <Header user={user}/>
-      {error ? <NotFound err={error}/> :
+      {error ? <NotFound err={error} resetError={resetError}/> :
```
Comment (chrissycooper) [inline client/src/Components/App/App.js:73]: BRILLIANT

File: client/src/Components/NotFound/NotFound.js (hunk 4 above)
Comment (chrissycooper) [inline client/src/Components/NotFound/NotFound.js:25]: thank you for adding it to the propTypes too!

**Review by chrissycooper — APPROVED** [review]

**Review by jfogiato — APPROVED** [review]

**Pull request merged**